composer/callbacks/mlperf.py

```diff
@@ -0,0 +1,345 @@
+
+    .. note::
+
+        MLPerf submissions require clearing the system cache prior to any training run.
+        By default, this callback does not clear the cache, as that is a system specific
+        operation. To enable cache clearing, and thus pass the mlperf compliance checker,
+        provide a ``cache_clear_cmd`` that will be executed with ``os.system``.
+
+    Args:
+        root_folder (str): The root submission folder
+        index (int): The repetition index of this run. The filename created will be
+            ``result_[index].txt``.
+        benchmark (str, optional): Benchmark name. Currently only ``resnet`` supported.
+        target (float, optional): The target metric before the mllogger marks the stop
+            of the timing run. Default: ``0.759`` (resnet benchmark).
+        division (str, optional): Division of submission. Currently only ``open`` division supported.
+        metric_name (str, optional): name of the metric to compare against the target. Default: ``Accuracy``.
+        metric_label (str, optional): label name. The metric will be accessed via ``state.current_metrics[metric_label][metric_name]``.
+        submitter (str, optional): Submitting organization. Default: MosaicML.
+        system_name (str, optional): Name of the system (e.g. 8xA100_composer). If
+            not provided, system name will default to ``[world_size]x[device_name]_composer``,
+            e.g. ``8xNVIDIA_A100_80GB_composer``.
+        status (str, optional): Submission status. One of (onprem, cloud, or preview).
+            Default: ``"onprem"``.
+        cache_clear_cmd (str, optional): Command to invoke during the cache clear. This callback
+            will call ``os.system(cache_clear_cmd)``. Default is disabled (None)
+    """
+
+    def __init__(
+        self,
+        root_folder: str,
+        index: int,
+        benchmark: str = 'resnet',
+        target: float = 0.759,
+        division: str = 'open',
+        metric_name: str = 'Accuracy',
+        metric_label: str = 'eval',
+        submitter: str = "MosaicML",
+        system_name: Optional[str] = None,
+        status: str = "onprem",
+        cache_clear_cmd: Optional[str] = None,
+    ) -> None:
+
+        require_mlperf_logging()
+
+        if benchmark not in BENCHMARKS:
+            raise ValueError(f"benchmark: {benchmark} must be one of {BENCHMARKS}")
+        if division not in DIVISIONS:
+            raise ValueError(f"division: {division} must be one of {DIVISIONS}")
+        if status not in STATUS:
+            raise ValueError(f"status: {status} must be one of {STATUS}")
+
+        self.mllogger = mllog.get_mllogger()
+        self.target = target
+        self.system_name = system_name
+        self.benchmark = benchmark
+        self.root_folder = root_folder
+        self.metric_name = metric_name
+        self.metric_label = metric_label
+
+        system_desc = get_system_description(submitter, division, status, system_name)
```

Comment:
Can this and everything below be moved to run on `Event.INIT` rather than on `__init__`? It is unknown when the callback will be constructed (and could be before distributed is initialized).

Reply:
Thanks for the suggestion -- not handling this for reasons described previously (e.g. debuggability and readability). Intended use should be with `composer` command.

Reply:
Distributed isn't configured until the trainer is constructed, so I'm pretty sure the `dist.barrier()` below will error out outside of tests. (In tests, we configure distributed _before_ any tests are run, so that is why they would be passing on Jenkins.) Can you please move where this code is run to Event.INIT?